### ISSUE TITLE
subnet & vnet peering maps should have a 'name' property, rather than use each.key

### ISSUE THREAD

**Issue by kewalaka** [Azure/terraform-azurerm-avm-res-network-virtualnetwork#65]
I don't think it is appropriate to use each.key as the name of the resource (though, it does make for less code)

https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/9df85d3feae34b65d1caedc9fb88dfd926dc75d3/main.subnet.tf#L7

If I'm reading the spec correctly, I believe the keys should be static & arbitrary, not form the name of the resource:

Ref: https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr7---category-code-style---count--for_each-use

i.e. the above example would look like:

```terraform
resource "azurerm_subnet" "subnet" {
  for_each = var.subnets

  address_prefixes   = each.value.address_prefixes
  name               = each.value.name
 //etc
}
```

& then the example would be specified like this:

```terraform
module "vnet_1" {
  source              = "../../"
  resource_group_name = azurerm_resource_group.example.name

  subnets = {
    subnet0 = {
      name             = module.naming.subnet.name_unique
      address_prefixes = ["192.168.0.0/16"]
    }
  }
```

This needs correction here too:
https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/9df85d3feae34b65d1caedc9fb88dfd926dc75d3/main.subnet.tf#L43

Separately, I don't think the peering should have a hard-coded prefix, and this too should use the name:

https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/9df85d3feae34b65d1caedc9fb88dfd926dc75d3/main.tf#L33

**Comment by matt-FFFFFF** [comment]
I think this is the root cause of #75

**Comment by jaredfholgate** [comment]
Resolved in v0.2.0
